### PR TITLE
Add windows icon support in core and more

### PR DESCRIPTION
**Pull request by dhaavi**
- Add support for getting binary icon and name from exe on Windows
- Update references to GenerateBinaryNameFromPath
- Update deps
- Remove security level from profile
- Remove CodeQL workflow

### CODE REVIEW

**Review by ppacher — DISMISSED** [review]

File: process/tags/appimage_unix.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (h *AppImageHandler) CreateProfile(p *process.Process) *profile.Profile {
 	if tag, ok := p.GetTag(appImagePathTagKey); ok {
 		return profile.New(&profile.Profile{
 			Source:              profile.SourceLocal,
-			Name:                osdetail.GenerateBinaryNameFromPath(p.Path),
+			Name:                icons.GenerateBinaryNameFromPath(p.Path),
 			PresentationPath:    p.Path,
```
[the remark this answers is not in the document]
Comment (dhaavi) [reply process/tags/appimage_unix.go:128]: Yes, I know.
It started with just icons, but names are so closely related, but I couldn't come up with an actual good name for the package. So I decided to leave it that way until we have better name. Recommendations?

File: profile/endpoints/endpoint-asn.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/safing/portmaster/intel"
 )
 
-var asnRegex = regexp.MustCompile("^(AS)?[0-9]+$")
+var asnRegex = regexp.MustCompile("^AS[0-9]+$")
```
[the remark this answers is not in the document]
Comment (dhaavi) [reply profile/endpoints/endpoint-asn.go:13]: Yes, this is difficult.

But, the issue is that the parser expects the "AS" prefix. Right now, "1234" will match AS34. So any rule that is really entered that way will in no circumstances meet the expectations of the user.

If rules break, there will be warnings, but Portmaster will still start.

Unfortunately we currently cannot decide if we are checking a new rule or parsing a rule from disk/DB. So a slow migration is hard.

File: process/tags/appimage_unix.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ppacher) [reply process/tags/appimage_unix.go:128]: what about `procmeta` ? or `binutils` ?

File: profile/endpoints/endpoint-asn.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ppacher) [reply profile/endpoints/endpoint-asn.go:13]: Got it :)

**Pull request merged**